src/movies.js

```diff
@@ -1,25 +1,117 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    let directors = [];
+    directors = moviesArray.map((element)=> { 
+        return element.director;
+    });
+    return directors;
+ }
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let totalTimes = 0;
+    if (moviesArray.length === 0) {
+        return 0;
+    }
+
+    const spielbegDramaMovies = moviesArray.filter((element)=>{
+        if(element.director === 'Steven Spielberg' && element.genre.includes('Drama')){
+            totalTimes++;
+        }
+    })
+
+    return totalTimes;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0) {
+      return 0;
+    }
+        
+    const scores = moviesArray.filter(movie => typeof movie.score === 'number');
+        
+    if (scores.length === 0) {
+       return 0;
+    }
+        
+    const total = scores.reduce((sum, movie) => sum + movie.score, 0);
+    const average = total / scores.length;
+        
+    return Math.round(average * 100) / 100;
```

Comment:
In this iteration you're not passing the last test because you're excluding the movies that don't have score, and what they want is for the movies that don't have score you count it as 0:

```
function scoresAverage(moviesArray) {
  if (moviesArray.length === 0) {
    return 0;
  }

  const total = moviesArray.reduce((sum, movie) => {
    if (!movie.score) {
      return sum;
    }
    return sum + movie.score;
  }, 0);

  const average = total / moviesArray.length;
  return Math.round(average * 100) / 100;
}
```